yarn-project/circuit-types/src/l2_block.ts

```diff
@@ -740,28 +454,17 @@ export class L2Block {
   //     `newL1ToL2Messages: ${inspectFrArray(this.newL1ToL2Messages)}`,
   //   ].join('\n');
   // }
+}
 
-  /**
-   * Computes logs hash as is done in the kernel and app circuits.
-   * @param logs - Logs to be hashed.
-   * @returns The hash of the logs.
-   * Note: This is a TS implementation of `computeKernelLogsHash` function in Decoder.sol. See that function documentation
-   *       for more details.
-   */
-  static computeKernelLogsHash(logs: TxL2Logs): Buffer {
-    const logsHashes: [Buffer, Buffer] = [Buffer.alloc(32), Buffer.alloc(32)];
-    let kernelPublicInputsLogsHash = Buffer.alloc(32);
-
-    for (const functionLogs of logs.functionLogs) {
-      logsHashes[0] = kernelPublicInputsLogsHash;
-      logsHashes[1] = functionLogs.hash(); // privateCircuitPublicInputsLogsHash
-
-      // Hash logs hash from the public inputs of previous kernel iteration and logs hash from private circuit public inputs
-      kernelPublicInputsLogsHash = sha256(Buffer.concat(logsHashes));
+function calculateNumTxsFromNullifiers(nullifiers: Fr[]) {
```

Comment:
Once again unrelated to your work but I think it would be cleaner if we nuked the `L2Block.numberOfTxs` property and replaced it with `get numberOfTxs` which would contain the implementation of `calculateNumTxsFromNullifiers.

Would you agree? It seems overly complicated to have a property on the block which is computed and set in constructor when computing it is cheap.